app.go

```diff
@@ -404,16 +417,16 @@ func HandleAction(action interface{}, context *Context) error {
 	if len(vals) == 0 {
 		fmt.Fprintln(os.Stderr,
 			"DEPRECATED Action signature.  Must be `cli.ActionFunc`")
-		return nil
+		return err
```

Comment:
I think this branch will only allow `nil`, yep.  I think I was being overly zealous with the `defer` impl above.